ui/src/features/Editor/components/LeftPanel/index.tsx

```diff
@@ -28,46 +40,112 @@ const LeftPanel: React.FC<Props> = ({
   onNodesAdd,
   isMainWorkflow,
   hasReader,
+  // onNodeDoubleClick,
+  selected,
 }) => {
   const t = useT();
   const { workspaceId } = useParams({ strict: false });
   const [selectedTab, setSelectedTab] = useState<Tab | undefined>();
+  const { fitView, getZoom, zoomTo } = useReactFlow();
+  const [previousZoom, setPreviousZoom] = useState<number | undefined>(
+    undefined,
+  );
+  const [nodeId, setNodeId] = useState<string | undefined>(undefined);
 
-  const [_content, setContent] = useState("Admin Page");
+  useEffect(() => {
+    if (previousZoom !== undefined && selected === undefined) {
+      zoomTo(previousZoom, { duration: 400 });
+    }
+  }, [previousZoom, selected, zoomTo]);
+
+  useEffect(() => {
+    if (selected && nodeId) {
+      setPreviousZoom(getZoom());
+    }
+
+    if (!isOpen) {
+      setNodeId(undefined);
+    }
+  }, [selected, isOpen, getZoom, nodeId]);
 
   useEffect(() => {
     if (!isOpen && selectedTab) {
       setSelectedTab(undefined);
     }
   }, [isOpen, selectedTab]);
 
+  const createTreeDataItem = (type: string, icon: Icon, name?: string) => {
```

Comment:
Move this function out of the main component so it renders only once and cleans things up a bit